### DISCUSSION TITLE
MFS chunk servers data migration

### DISCUSSION THREAD

**Discussion by aaafei123** [moosefs/moosefs#529] [Q&A]
Hi all：

We have the following requirements:

Here we have 5 chunk servers, like below:
    - mfschunk server01:
        /mnt/disk01     1TB     85% used
        /mnt/disk02     1TB     85% used
    - mfschunk server02:
        /mnt/disk01     1TB     85% used
        /mnt/disk02     1TB     85% used
    - mfschunk server03:
        /mnt/disk01     4TB     15% used
        /mnt/disk02     4TB     15% used
    - mfschunk server04:
        /mnt/disk01     4TB     15% used
        /mnt/disk02     4TB     15% used
    - mfschunk server05:
        /mnt/disk01     4TB     15% used
        /mnt/disk02     4TB     15% used

Servers Distribution:
    1) AWS Cloud: mfschunk server01-02
    2) Ali Cloud: mfschunk server03-05

Now we want to move all data from mfschunk server01-02 to mfschunk server03-05, but we found there are some issues:
if we marked one disk for removal on chunk server, like server01's: */mnt/disk02, then the capacity of disk on mfschunk server02 will exceed 98% even 100%, the speed of files read/write operation are very slow, all mfs clients can barely read and write files properly.

btw: the version of mfs is v3.0.94

So how should we do? 
thanks a lot.

**Comment by oszafraniec** [comment]
@aaafei123 do those chunkservers have classes assigned? do you use classes or just goals while defining replication rules? where are clients located, AWS or Ali or both?

**Reply by aaafei123** [reply]
@oszafraniec thank you:)
we just use goals and mfs clients locate both on AWS and Ali.

**Reply by oszafraniec** [reply]
what goals? if you have data with goal 4, then removing server02 will end up filling server01 with 4-th copy that was till now on server02. Mayby this is why it is filling up.

Read about classes, if you have multi-location cluster, classes are a must in my opinion. You can replicate data betwenn location (1 copy in AWS, 1 copy in Ali) and also tell clients to use servers locally first (client in Ali, use chunk copy from Ali, if now go to AWS).

https://moosefs.com/Content/Downloads/moosefs-storage-classes-manual.pdf

Classes can be also used to migrate data. Make class "AWS" and "Alli" with same goal. Then change the class of files from AWS to Ali and see how MFS will move chunks where they should be without downtime.

**Reply by aaafei123** [reply]
@oszafraniec okie, thanks again:) I read the manual and try it later. I think I should upgrade my MFS first and then try your method  mentioned above.

**Comment by chogata** [comment]
You wrote that your MooseFS instance version is 3.0.94. Between this version and the current one there are a LOT of improvements and bug fixes. Why do you keep such old version? Is it possible to upgrade your instance to 3.0.117? I'm 99% sure your problem with mark for removal will solve itself and even if not, we will be much better equipped to help you :)

**Reply by aaafei123** [reply]
thanks:) I try to upgrade my MFS later.
